src/lib/components/TableOfContents/index.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {html} from 'lit-element';
+import {BaseElement} from '../BaseElement';
+import {stringToBoolean} from '../../utils/string-to-boolean';
+
+/**
+ * Element that renders table of contents.
+ * @extends {BaseElement}
+ * @final
+ */
+class TableOfContents extends BaseElement {
+  static get properties() {
+    return {
+      title: {type: String},
+      titleId: {type: String, attribute: 'title-id'},
+      opened: {type: Boolean, converter: stringToBoolean, reflect: true},
```

Comment:
You shouldn't need a converter, lit will do it for you.

Here's an example:
https://glitch.com/edit/#!/adjoining-erratic-cake?path=index.html%3A7%3A25

A boolean attribute is true if it's present on the html and false if it is absent.
e.g.

```html
<my-foo opened> // opened is true
<my-foo opened=""> // opened is true
<my-foo> // opened is false
```

Reply:
So you should be able to do something like:

```js
updated(changedProperties) {
  if (changedProperties.has('opened')) {
    const opened = changedProperties.get('opened');
    if (opened) {
      this.scrollSpy();
      document.addEventListener('touchmove', this.scrollSpy, {passive: true});
      document.addEventListener('scroll', this.scrollSpy, {passive: true});
    } else {
      document.removeEventListener('touchmove', this.scrollSpy);
      document.removeEventListener('scroll', this.scrollSpy);
    }
  }
}
```

Reply:
So the converter works for cases when `<my-foo opened="false">`.

As for the `updated` method I still think we should have the `open` and `close` methods. Especially the `close` one. The document event listened can exist after the element is removed, so I call the `close` when the element is removed.